besmarts-core/python/examples/07_besmarts_fit.py

```diff
@@ -161,21 +204,53 @@ def run():
     """
 
     # 1 Build the dataset and FF
-    d = make()
-    csys = load_sage_csys()
-    gdb = new_gdb(d)
-    psys = fits.gdb_to_physical_systems(gdb, csys)
+    example_training_data: dict[str, list[dict]] = make()
+
+    # load the Sage SMIRNOFF force field into a
+    # ``molecular_mechanics.chemical_system``.
+    #  The ``chemical_system`` has the following models in order:
+    # bonds, angles, torsions, outofplanes, electrostatics, vdw.
+    # Each model is an instance of ``molecular_mechanics.chemical_model``.
+    csys: mm.chemical_system = load_sage_csys()
+    gdb: assignments.graph_db = new_gdb(example_training_data)
+
+    # set up the physical systems
+    # a molecular_mechanics.physical_system is a collection of
+    # ``molecular_mechanics.physical_model`` objects.
+    # A physical model is the functional form that can be evaluated
+    # as a function of positions
+    psys_dict: dict[int, mm.physical_system] = fits.gdb_to_physical_systems(
+        gdb, csys
+    )
 
     # 1 Configure the fitting strategy
+    # models should be a dictionary where the keys are the indices
+    # of each ``molecular_mechanics.chemical_model`` object
+    # in the ``csys``. Therefore,
+    # the 0 index here refers to the bond model.
+    # if we wanted to train an angle, we would need
+    # to set the key to 1.
     models = {0: ["b4"]}
-    strat = fits.forcefield_optimization_strategy_default(csys, models=models)
-    co = fits.chemical_objective
+
+    # this sets up the default optimization strategy for the fit.
+    # It uses the ``models`` dictionary to determine which models
+    # to fit in the ``csys.models`` list.
+    # default settings are also applied to splitting, extending, perception
+    # configs etc.
+    strategy = fits.forcefield_optimization_strategy_default(
+        csys, models=models
+    )
+    chemical_objective_function = fits.chemical_objective
 
     # 2. Configure the objective tiers
     final = fits.objective_tier()
+    # set the objectives for the final tier
+    # each value is an ``objective_config``
     final.objectives = {
         0: fits.objective_config_position(
```

Comment:
I'm not too sure what these keys represent -- do they represent order, or do they maybe correspond with the `assignments.POSITIONS` index?

Reply:
The objective keys are completely arbitrary ints; they represent order. I did it more for the case of when the the objective list gets split up they maintain their "index"

Reply:
They have no _almost_ no impact on how the objective is computed, other than the order they are added (FIFO; put expensive things first). I haven't put much thought into the objective keys, but I've been assuming ints. Looking through the code, I think the only part that uses the objective key is when I break up the particular computation into pieces, I keep track of ownership using the key. So, I would guess anything hashable is usable as a key. For example I have this in `optimizers_scipy`:
```
tasks[(n, (i, (0,)))] = task
```
where `n` is the iteration number, `i` is the objective key in question, and (0,) refers to no parameter deltas (e.g. the f(x0) in the numerical derivatives. This is mostly useful for Hessians, where a molecule can have 100s of deltas to compute, and I break them into pieces and distribute them for computation. I collect the results based on the objective key (`i` above). That said, they shouldn't need to be continuous, just something hashable to organize computations with.